src/wp-admin/includes/class-wp-automatic-updater.php

```diff
@@ -511,6 +511,9 @@ public function update( $type, $item ) {
 				&& ( 'up_to_date' === $upgrade_result->get_error_code()
 					|| 'locked' === $upgrade_result->get_error_code() )
 			) {
+				// Allow visitors to browse the site again.
+				$upgrader->maintenance_mode( false );
```

Comment:
Theoretically and logically, this change makes sense. 

It bothers me that I'm not (yet) able to reproduce the issue of the maintenance lock.